### PR TITLE
How the book would read if programs were renamed to services

### PR DESCRIPTION
**Pull request by garious**
Here's how the programming model would be described if programs were renamed to services. An interactive program naturally inherits the name *escrow service*. Not mentioned here, an example of a stateless service would be a *notary service*.

### CODE REVIEW

File: book/src/programs.md [hunk 1]
```diff
@@ -2,65 +2,67 @@
...
+subsequent transactions.
 
-A program may be written in any programming language that can target the
+A service may be written in any programming language that can target the
 Berkley Packet Filter (BPF) safe execution environment. The Solana SDK offers
-the best support for C programs, which is compiled to BPF using the [LLVM
-compiler infrastructure](https://llvm.org). Alternatively, a client might
-choose to bypass LLVM and use Python, Lua or C++ to generate BPF directly via
-the [BPF Compiler Collection](https://github.com/iovisor/bcc) (BCC).
+the best support for services written in C, which is compiled to a plugin using
```
Comment (mvines) [inline book/src/programs.md]: `s/plugin/shared object/g`?

Comment (mvines) [reply book/src/programs.md]: Hmm, actually why do we even care to distinguish between "service" and "plugin"/"shared object".   We're splitting split hairs at this point.

**Comment by mvines** [comment]
Overall I think the `s/program/service/g` here reads ok.  I'm still finding "service" to be not substantially better than "program" but close enough for me that either way feels like a shrug.

**Comment by aeyakovenko** [comment]
My feel is that service describes active state. Programs are just stateless code blobs. And using the escrow and notary words is really confusing.

**Comment by garious** [comment]
Bad bang for buck, closing.

**Pull request closed without merging**